src/admin/frontend/components/article-edit-panel.js

```diff
@@ -215,52 +270,49 @@ class ArticleSidebar extends Component {
               <div className="form-group">
                   <label htmlFor="slug-input">Slug</label>
                   <input id="slug-input" name="slug" type="text" className="form-control" placeholder="Slug"
-                      defaultValue={article.slug}
-                      onChange={ e => this.props.updateArticle('slug', e.target.value) } />
+                      value={article.slug}
+                      onChange={ e => this.context.updateArticle('slug', e.target.value) } />
               </div>
               
               <div className="form-group">
                   <label htmlFor="response-to-input">Response To</label>
-                  <ArticleGroup name="parent" query={{
-                    _id: article.parent,
-                    limit: 1,
-                  }}>
-                      { articles => {
-                        const parent = articles[0] || {};
-                        return (
-                          <TypeaheadInput
-                            key={parent._id} type="text" className="form-control" placeholder="Response To..." defaultValue={parent.title}
-
-                            typeaheadConfig={{
-                              hint: true,
-                              highlight: true,
-                              minLength: 1,
-                              display: 'title',
-                            }}
-
-                            createBloodhoundConfig={function(Bloodhound) {
-                              return new Bloodhound({
-                                datumTokenizer: Bloodhound.tokenizers.obj.whitespace('title'),
-                                queryTokenizer: Bloodhound.tokenizers.whitespace,
-                                remote: {
-                                  url: '/admin/articles/search?title=%QUERY',
-                                  wildcard: '%QUERY',
-                                },
-                              });
-                            }}
-
-                            target={this.refs['response-id-field']}
-                            targetField="_id"
-                          />
-                        );
-                      }}
-                  </ArticleGroup>
+                  
+                  <ResponseToInput article={article}>
+                    {({parent}) => {
+                      return (
+                        <TypeaheadInput key={parent._id} type="text" className="form-control" placeholder="Response To..." defaultValue={parent.title}
+                     
+                          typeaheadConfig={{
+                            hint: true,
+                            highlight: true,
+                            minLength: 1,
+                            display: 'title',
+                          }}
+                     
+                          createBloodhoundConfig={function(Bloodhound) {
```

Comment:
We give it a url to query article titles, and then it will automatically do all the caching and throttling so it doesn't have a request everytime you time a letter